text/0000-arguments.md

```diff
@@ -120,16 +125,18 @@ The `launcher` currently has a three step command resolution flow:
 
 1.  0 args: execute the default process type, where the default process type is `${CNB_PROCESS_TYPE:-web}`
 1.  1 arg:  if `$1` matches a process type, execute that process type
-1.  N args: execute `$1` as the command with `${@:2}` as the arguments appended to that command
+1.  N args if `$1` is `--`: directly execute `$2` as the command with `${@:3}` as the arguments appended to that command
+1.  N args: execute `$1` as the command with `${@:2}` as the arguments appended to that command in a `bash` shell
 
-This proposal is to augment that command resolution flow to:
+This proposal change the command resolution flow to:
 
 1.  If `CNB_PROCESS_TYPE=override`
     1. 0 args: fail
-    1. N args: execute `$1` as the command with `${@:2}` as the arguments
+    1.  N args if `$1` is `--`: directly execute `$2` as the command with `${@:3}` as the arguments appended to that command
+    1.  N args: execute `$1` as the command with `${@:2}` as the arguments appended to that command in a `bash` shell
 1.  Else
     1. 0 args: execute the process type `${CNB_PROCESS_TYPE:-web}`
-    1. N args: Execute process type `${CNB_PROCESS_TYPE:-web}`, after appending `"$@"` to the arguments
+    1. N args: Execute process type `${CNB_PROCESS_TYPE:-web}`, after appending `$@` (all positional arguments) to the process arguments
```

Comment:
Why we don't allow the `--` option as we do when `CNB_PROCESS_TYPE=override`?

Reply:
When you are appending arguments to an existing process-type `direct=true` vs. `direct=false` is already defined for the process. It wouldn't be safe to change the value of `direct` for a buildpack-provided process.